src/modules/ChannelList/dux/reducers.js

```diff
@@ -114,79 +122,94 @@ export default function reducer(state, action) {
     }
     case actions.ON_USER_LEFT: {
       const { channel, isMe } = action.payload;
-      if (state.channelListQuery) {
-        if (filterChannelListParams(state.channelListQuery, channel, state.currentUserId)) {
-          const filteredChannels = getChannelsWithUpsertedChannel(state.allChannels, channel);
-          const nextChannel = (isMe && (channel?.url === state.currentChannel?.url))
-            ? filteredChannels[0]
-            : state.currentChannel;
-          return {
-            ...state,
-            currentChannel: state.disableAutoSelect ? null : nextChannel,
-            allChannels: filteredChannels,
-          };
+      const {
+        allChannels,
+        currentUserId,
+        currentChannel,
+        channelListQuery,
+        disableAutoSelect,
+      } = state;
+      let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
+      let nextChannel = null;
+      if (channelListQuery) {
+        if (filterChannelListParams(channelListQuery, channel, currentUserId)) {
+          // Good to [add to/keep in] the ChannelList
+          nextChannels = getChannelsWithUpsertedChannel(allChannels, channel);
```

Comment:
why upsert removed channel again here?

Reply:
It's not removed,
```
let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
```
This is just default value